spyking-circus/spyking-circus
# Move to GitHub actions

This replaces travis and appveyor by [GitHub Actions](https://github.com/spyking-circus/spyking-circus/actions) and simplifies the whole testing/deployment setup along the way.

There are four "workflow" files (in principle it could be a single one, but you can only restart complete workflows, so it can be useful to have them split up):
<dl>
<dt>Build and publish PyPI packages</dt>
<dd>Builds the source package and the wheel, and tries to install from these packages via pip</dd>
<dt>Tests on Linux/Windows/OS X</dt>
<dd>Builds the conda package on the respective OS for Python 3.6, 3.7, 3.8. The only test at the moment is trying to import the package and running `spyking-circus -h`.</dd>
</dl>

I did not bother with Python 2 support for now, do you still want to keep it? If not, it might be worth getting rid of the `2to3.py` clutch as well... Oh, and on Windows we cannot build a conda package for Python 3.8 due to dependency issues, but it might just be because there is no `mpi4py` package for Python 3.8 in the `intel` channel?

All of these workflows should upload the respective PyPI/conda packages if you tag a new release, but obviously I did not test this yet :) It also still needs a bit additional configuration for the tokens:
* Create an API token on PyPI: https://pypi.org/help/#apitoken
* Add it as a "secret" named `pypi-password` to the repository: https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets#creating-encrypted-secrets-for-a-repository
* Create a token for Anaconda with "scope" for read and write access to the API: https://anaconda.org/spyking-circus/setting/access
* Add this token as another secret, called `ANACONDA_TOKEN`

I also wonder (but this would be a separate PR once the general setup is working) whether it would make sense to switch to [`versioneer`](https://github.com/python-versioneer/python-versioneer). If this works, it would mean that you'd never have to update the version number manually, just tagging the release would be enough (and even users that use `spyking-circus` directly from github would have a meaningful version number).
